kaniko/kaniko.yaml

```diff
@@ -9,13 +9,17 @@ spec:
   - name: DOCKERFILE
     description: Path to the Dockerfile to build.
     default: /workspace/Dockerfile
+  - name: CONTEXT
+    description: Build Context
+    default: /workspace
 
   steps:
   - name: build-and-push
     image: gcr.io/kaniko-project/executor
     args:
     - --dockerfile=${DOCKERFILE}
     - --destination=${IMAGE}
+    - --context=${CONTEXT}
```

Comment:
I was thinking whether it is a good idea to hide /workspace from the end user. Instead of setting /workspace/<real path>, we can directly set <real path> or /<real path> to CONTEXT. Instead of setting /workspace/<path to Dockerfile>, we can directly set <path to Dockerfile> or /<path to Dockerfile> to dockerfile.

If you like this idea, the only change is to change this line into:
--context=/workspace/${CONTEXT}

Reply:
yeah, the workspace bit confused me as a beginer (still am).
i don't mind altering it but isn't modifying  context going somone currently using this this because the existing defaults specifies `/workspace` already here
https://github.com/knative/build-templates/blob/master/kaniko/kaniko.yaml#L11

..and not that it makes a difference but the other docs forces using a `/workspace` too , eg:
  https://github.com/GoogleContainerTools/kaniko#running-kaniko-in-gvisor

Reply:
I had some similar PRs for the tektoncd pipeline to shield the /workspace: https://github.com/tektoncd/catalog/pull/32  I guess we can apply here as well.

To users, it is more straightforward to set CONTEXT to a dir directly under the home, regardless of /workspace.

Reply:
one thing to note is the regular non-template build conformance in that the paths will be different.

so if i define a regular build
```yaml
apiVersion: build.knative.dev/v1alpha1
kind: Build
metadata:
  name: kaniko-build
spec:
  serviceAccountName: default
  source:
    git:
      url: https://github.com/salrashid123/istio_helloworld.git
      revision: master
  steps:
  - name: build-and-push
    image: gcr.io/kaniko-project/executor:v0.9.0
    args:
    - --dockerfile=./Dockerfile
    - --destination=gcr.io/mineral-minutia-820/istioinit:3
    - --context=/workspace/nodeapp
    env:
    - name: DOCKER_CONFIG
```

i'd have to specify the path ` - --context=/workspace/nodeapp`

but if `/workspace` is already accounted for me within a template, my migration to using the template could catch users off guard..

(eg, i'd have remember to remove `/workspace/`).

- if i already accounted for the path within the template i'd do 
- A:

```yaml
apiVersion: build.knative.dev/v1alpha1
kind: Build
...
  template:
    name: kaniko
    arguments:
    - name: IMAGE
      value: gcr.io/mineral-minutia-820/istioinit:3
    - name: DOCKERFILE
      value: ./Dockerfile
    - name: CONTEXT
      value: /nodeapp
```


but without doing that its

- B

```yaml
apiVersion: build.knative.dev/v1alpha1
kind: Build
...
  template:
    name: kaniko
    arguments:
    - name: IMAGE
      value: gcr.io/mineral-minutia-820/istioinit:3
    - name: DOCKERFILE
      value: ./Dockerfile
    - name: CONTEXT
      value: /workspace/nodeapp
```

(B) is closer to what you'd use if you were just doing the build from scratch

Reply:
@mattmoor @ImJasonH   

any thoughts on this PR?   The main thing is related to the context we should use here.  From what i see, the consistent way is to keep the `/workspace/` bit